### PR TITLE
`String()` method instead of the `string` func

### PR DESCRIPTION
**Pull request by wulianglongrd**
Two minor optimizations
1. two identical variables, one for `if` check and one for `return` are hard to read.
2. call the `String()` method instead of the `string` func

### CODE REVIEW

File: istioctl/cmd/describe.go [hunk 1]
```diff
@@ -293,7 +293,7 @@ func httpRouteMatchSvc(vs *clientnetworking.VirtualService, route *v1alpha3.HTTP
 	mismatchNotes := []string{}
 	match := false
 	for _, dest := range route.Route {
-		fqdn := string(model.ResolveShortnameToFQDN(dest.Destination.Host, config.Meta{Namespace: vs.Namespace}))
+		fqdn := model.ResolveShortnameToFQDN(dest.Destination.Host, config.Meta{Namespace: vs.Namespace}).String()
```
Comment (howardjohn) [inline istioctl/cmd/describe.go:296]: I would encourage you to focus on real bugs and issues. Changes like this do not benefits users and just add maintenance cost

Comment (wulianglongrd) [reply istioctl/cmd/describe.go:296]: Got it, these were changed while reading the code. Didn't realize it took up a lot of the maintainer's time, so sorry, won't be doing it again.

Comment (hzxuzhonghu) [reply istioctl/cmd/describe.go:296]: I believe the previous way is already very clear

Comment (wulianglongrd) [reply istioctl/cmd/describe.go:296]: @hzxuzhonghu hard to read refers to [this](https://github.com/istio/istio/pull/43147/files#diff-15fbe3ad55d969d0f1a0c8d41405acdead478748d4520bc37a87c196d22d1859) modification, here is to replace the `string` function with the `String` method (many place use `String` method convert `host.Name` to string).

I'm a golang beginner coming from java and trying to become a idiomatic golanger. I will think more before submitting my next PR.

Thank you everyone.

**Pull request closed without merging**